app/screens/AppNavigator/screens/Settings/screens/ConfirmPinScreen.tsx

```diff
@@ -0,0 +1,83 @@
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { ActivityIndicator, ScrollView } from 'react-native'
+import { Logging } from '../../../../../api'
+import { MnemonicEncrypted } from '../../../../../api/wallet'
+import { MnemonicWords } from '../../../../../api/wallet/mnemonic_words'
+import { Text, View } from '../../../../../components'
+import { PinTextInput } from '../../../../../components/PinTextInput'
+import { useNetworkContext } from '../../../../../contexts/NetworkContext'
+import { useWalletPersistenceContext } from '../../../../../contexts/WalletPersistenceContext'
+import { tailwind } from '../../../../../tailwind'
+import { translate } from '../../../../../translations'
+import { SettingsParamList } from '../SettingsNavigator'
+
+type Props = StackScreenProps<SettingsParamList, 'ConfirmPinScreen'>
+
+export function ConfirmPinScreen ({ route }: Props): JSX.Element {
+  const { network } = useNetworkContext()
+  const { setWallet } = useWalletPersistenceContext()
+  const { pin, words } = route.params
+
+  const [newPin, setNewPin] = useState('')
+  const [invalid, setInvalid] = useState<boolean>(false)
+  const [spinnerMessage, setSpinnerMessage] = useState<string>()
+
+  function verifyPin (input: string): void {
+    if (input.length !== pin.length) return
+    if (input !== pin) {
+      setNewPin('')
+      setInvalid(true)
+      return
+    } else {
+      setInvalid(false)
+    }
+
+    const copy = { words, network, pin }
+    setSpinnerMessage(translate('screens/PinConfirmation', 'Encrypting wallet...'))
+    setTimeout(() => {
+      MnemonicEncrypted.toData(copy.words, copy.network, copy.pin)
+        .then(async encrypted => {
+          await MnemonicWords.encrypt(words, pin)
+          await setWallet(encrypted)
+
+          // if (biometricEnrolled) BiometricProtectedPasscode.set(pin)
```

Comment:
Biometric and passcode changes are individual settings. Update one will update another, but not tweak them together.